tests/test_cleanup.py

```diff
@@ -69,15 +70,29 @@ class TestCleanup:
     @pytest.mark.parametrize(
         "storage_key, create_event_row_for_date", test_data,
     )
-    def test(
+    @patch("snuba.cleanup.current_time")
+    def test_main_cases(
         self,
+        current_time: MagicMock,
         storage_key: StorageKey,
         create_event_row_for_date: Callable[[datetime, Optional[int]], InsertBatch],
     ) -> None:
         def to_monday(d: datetime) -> datetime:
-            return d - timedelta(days=d.weekday())
-
-        base = datetime(1999, 12, 26)  # a sunday
+            rounded = d - timedelta(days=d.weekday())
+            return datetime(rounded.year, rounded.month, rounded.day)
+
+        # In prod the dates have hours/seconds etc.
+        prebase = datetime.utcnow()
+        base = datetime(
+            1999,
+            12,
+            26,
+            prebase.hour,
+            prebase.minute,
+            prebase.second,
+            prebase.microsecond,
+        )  # a sunday
```

Comment:
You could just update the hardcoded base time to include some hours, minutes, seconds. It's less verbose than using combination of hardcoded and actual time and merging them together